src/main/java/org/kiwiproject/beta/reflect/TypeInfo.java

```diff
@@ -0,0 +1,171 @@
+package org.kiwiproject.beta.reflect;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.collect.KiwiLists.first;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Value;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents either a simple type (e.g. Boolean or String) or a parameterized type (e.g. {@code List<String>}
+ * or {@code Map<String, Integer>}).
+ */
+@Value
+public class TypeInfo {
+
+    /**
+     * If a simple type, this is the entirety of the type information. If a parameterized type, this is the
+     * top-level type, e.g. List or Map.
+     */
+    Type rawType;
+
+    /**
+     * If a simple type, this will be empty. If a parameterized type, then this list contains the generic types.
+     * For example, a {@code List<String>} has one generic type, String, while a {@code Map<String, Integer>}
+     * contains two generic types, String and Integer.
+     */
+    List<Type> genericTypes;
```

Comment:
@sonatype-lift ignoreall

Lift, or the tools it invokes like ErrorProne, don't seem to understand code that uses Lombok. So, telling it to ignore all these since it is coming from code that Lombok is generating via `@Value`

Reply:
@sonatype-lift ignoreall